src/snmalloc/mem/metadata.h

```diff
@@ -685,7 +685,9 @@ namespace snmalloc
      */
     [[nodiscard]] SNMALLOC_FAST_PATH SlabMetadata* get_slab_metadata() const
     {
-      //SNMALLOC_ASSERT(get_remote() != nullptr);
+      // TODO Following assertion removed for client meta-data use case.
+      // Think about possible UAF scenarios.
+      // SNMALLOC_ASSERT(get_remote() != nullptr);
```

Comment:
The concern is that the client has retained a pointer to a free object for so long that the containing slab has been given back to the backend and said client is now asking for the metadata once associated with that free object?  At that point `meta` isn't even guaranteed to point to a metaslab, right?

Reply:
Yeah, this could do with some thinking about. What we might do to prevent a UAF being escalated.

Reply:
https://github.com/microsoft/snmalloc/pull/662/commits/a93cd15d740f450b48fc770929242a332bf82a47

So this checks if it is a backend meta-data.  There is a gap between the check and following pointers that could potentially be exploited to get an interesting write gadget, but this is a short window, and require control over interleavings to make reliable.